src/Types/OnChainData.cs

```diff
@@ -25,6 +29,11 @@ public sealed record OnChainData : IEquatable<OnChainData>
     /// <param name="bytes">Data represented by at most <see cref="MaxLength"/> bytes.</param>
     private OnChainData(byte[] bytes) => this._value = bytes;
 
+    /// <summary>
+    /// Gets the length (number of bytes) of the data.
+    /// </summary>
+    internal uint Length() => (uint)this._value.Length;
```

Comment:
It appears that we're continuously switching between different casts of uint and int. Wouldn’t it be better to just store as int and then only cast when interacting with the chain?

Reply:
I also dislike it. I made it uint as the `BytesLength` in CcdAmount, Transfer, AccountAddress, etc. are all of uint. So this is to keep consistency.

Reply:
What do you think is the best solution here?